### PR TITLE
Use solc.linkBytecode for linking

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/universal-dapp.js [hunk 1]
```diff
@@ -556,15 +557,7 @@ UniversalDApp.prototype.linkBytecode = function (contractName, cb) {
...
-    var libLabel = '__' + libraryName + Array(39 - libraryName.length).join('_');
-    var hexAddress = address.toString('hex');
-    if (hexAddress.slice(0, 2) === '0x') {
-      hexAddress = hexAddress.slice(2);
-    }
-    hexAddress = Array(40 - hexAddress.length + 1).join('0') + hexAddress;
-    while (bytecode.indexOf(libLabel) >= 0) {
-      bytecode = bytecode.replace(libLabel, hexAddress);
-    }
+    bytecode = solc.linkBytecode(bytecode, { [libraryName]: ethJSUtil.addHexPrefix(address.toString('hex')) });
```
Comment (chriseth) [inline src/universal-dapp.js]: Is it really `[libraryName]`?

Comment (axic) [inline src/universal-dapp.js]: `libraryName` is a literal, while `[libraryName]` will use the referenced value as key

**Comment by axic** [comment]
I'm not sure what is broken with tests, but I get seemingly unrelated issues:
> Testing if element <.active> contains text: "Untitled".  - expected "Untitled" but got: ""

**Comment by chriseth** [comment]
These test errors mean that either the application did not start up at all in the browser or some js did not initialize correctly.

**Comment by chriseth** [comment]
Please rebase

**Comment by chriseth** [comment]
Please rebase.

**Comment by axic** [comment]
This actually won't work in this form, because `linkBytecode` is not a static method of the solc wrapper - it needs a soljson loaded. We need to pass down the loaded compiler to the udapp in that case.

**Comment by axic** [comment]
Depends on https://github.com/ethereum/solc-js/pull/150

**Comment by axic** [comment]
@yann300 updated this, it should work, but haven't tested.

**Comment by axic** [comment]
@yann300 rebased!

**Comment by yann300** [comment]
moved to remix repo

**Pull request closed without merging**